4-storage/storage-loader/lib/snowplow-storage-loader/dashdb_loader.rb

```diff
@@ -0,0 +1,122 @@
+
+# Ruby module to support the load of Snowplow events into DashDB
+module Snowplow
+  module StorageLoader
+    module DashDBLoader
+
+    include Contracts
+
+    # Used to find the altered enriched events
+    RUN_PATTERN = /(run=[0-9\-]+\/atomic-events)/
+
+    # Loads the Snowplow atomic event files into DashDB table
+    #
+    # Parameters:
+    # +config+:: the configuration options
+    # +target+:: the configuration for this specific target
+    Contract Hash, Hash => nil
+    def load_atomic_events_into_dashdb(config, target, snowplow_tracking_enabled)
+      puts "Loading Snowplow events #{target[:name]} (DashDB instance)..."
+
+      run_folder = retrieve_run_folder(config, snowplow_tracking_enabled)
+
+      response_value = create_run_dataworks_transfer(config, target, run_folder)
+
+      if response_value != 200
+        if snowplow_tracking_enabled
+          Monitoring::Snowplow.instance.track_load_failed('Failure to transfer atomic events from S3 => DashDB, response #{response_value}.')
+        end
+      end
+
+      if snowplow_tracking_enabled
+        Monitoring::Snowplow.instance.track_load_succeeded()
+      end
+
+      nil
+    end
+    module_function :load_atomic_events_into_dashdb
+
+    private
+
+      # Call dataworks_proxy api endpoint with a request body that passes DashDB and S3 details.
+      # The endpoint will create the S3 => DashDB activity and also kick it off.
+      # ** Still waiting on DataWorks team to add new functionality where file pattern matching is allowed.
+      #
+      # Parameters:
+      # +config+:: the configuration options
+      # +target+:: the configuration for this specific target
+      # +run_folder+:: the retrieved run folder from shredded/good
+      Contract Hash, Hash => Num
+      def create_run_dataworks_transfer(config, target, run_folder)
+        url = URI.parse(target[dataworks_proxy])
+        req = Net::HTTP::Post.new(url)
+        req['Accept'] = 'application/json'
+        req.body = '{
+          "api_key":"#{target[:api_key]}",
+          "host":"#{target[:host]}",
+          "username":"#{target[:username]}",
+          "password":"#{target[:password]}",
+          "database":"#{target[:database]}",
+          "schema":"#{target[:schema]}",
+          "table":"#{target[:table]}",
+          "s3_region":"#{config[:aws][:s3][:region]}",
+          "s3_access_key":"#{config[:aws][:access_key_id]}",
+          "s3_secret_key":"#{config[:aws][:secret_access_key]}",
+          "s3_bucket":"#{config[:aws][:s3][:buckets][:shredded][:good]}/#{run_folder}"
+        }'
+
+        res = Net::HTTP.start(url.host, url.port, :use_ssl => url.scheme == 'https') { |http| http.request(req) }
+
+        res.value
+      end
+      module_function :create_run_dataworks_transfer
+
+      # Retrieve run folder from shredded/good. Max one folder in there at any given time.
+      #
+      # Parameters:
+      # +config+:: the configuration options
+      Contract Hash => String
```

Comment:
I'm actually not privy of Contracts so I was following redshift_loader as an example. From that class, in load_events_and_shredded_types, the Contract is Hash, Hash => nil, so I figured that snowplow_tracking_enabled is optional?